### ISSUE TITLE
Changing Session ID still gives same results [BUG]: 

### ISSUE THREAD

**Issue by nabgit** [pycaret/pycaret#3650]
### pycaret version checks

- [X] I have checked that this issue has not already been reported [here](https://github.com/pycaret/pycaret/issues).

- [X] I have confirmed this bug exists on the [latest version](https://github.com/pycaret/pycaret/releases) of pycaret.

- [X] I have confirmed this bug exists on the master branch of pycaret (pip install -U git+https://github.com/pycaret/pycaret.git@master).

### Issue Description

When using pycaret given same dataset, with same train, test and val sets the compare_models function always outputs the same results, even changing session_id does not change results. 

### Reproducible Example

```python
import numpy as np
from sklearn.metrics import roc_auc_score, confusion_matrix
from sklearn.model_selection import train_test_split
from pycaret.classification import *
import random

for j in range(num_sub_runs):
        print(f"\tSub-run {j+1}/{num_sub_runs}")
        random_seed = random.randint(1,10000)

        # Set up the data and environment
        model_setup = setup(data=df_train, test_data = df_val, target='diagnosis', fold=5, 
         fold_strategy ='stratifiedkfold', fix_imbalance=True, index=False, session_id=random_seed)

        # Train the single model/ensemble model
        model_single = compare_models(cross_validation= False, sort='AUC')
```

### Expected Behavior

![image](https://github.com/pycaret/pycaret/assets/110167436/f000f645-0244-47a0-8d56-8f9c5de98d23)
![image](https://github.com/pycaret/pycaret/assets/110167436/3177b895-0da3-491d-b6ff-d8d6015c861b)

### Actual Results

```python-traceback
Since I am running compare_models in a loop, changing session_id on same dataset with same train,test,val gives almost same results. I want different results in loop, please correct me if I am wrong. Thanks
```

### Installed Versions

<details>
pycaret-3.0.4
</details>

**Comment by ngupta23** [comment]
Session ID is just a seed and should not be treated as a hyperparameter in the model to expect different results. It should only be used for reproducibility.
